### ISSUE TITLE
Set-SCPItem hard lock on macOS

### ISSUE THREAD

**Issue by ChicagoJay** [darkoperator/Posh-SSH#445]
Running macOS 10.15.7 with PowerShell 7.2.1 and Posh-SSH 3.0.0

When I invoke `Set-SCPItem -ComputerName "SCPhost.FQDN" -Credential username /sourcepath/to/file -Destination /`, I am prompted for credentials, then pwsh locks up hard: ctrl-c, soft and hard resets don't get the prompt back.  Only way to reset is to kill the pwsh process.

I tried changing the source path delimiters to \ from /, but the result was identical.  Am I doing something wrong?

**Comment by ChicagoJay** [comment]
I understand that Macs have scp built into the pwsh library, but I am trying to use the same code for both Macs and Windows.  Do I really have to write two blocks and have the script choose, based on the OS from which is is run?

**Comment by MVKozlov** [comment]
The reason seems to be the same as in #349. (https://github.com/sshnet/SSH.NET/issues/745)
scp server does not support file time setting

**Comment by ChicagoJay** [comment]
> The reason seems to be the same as in #349. ([sshnet/SSH.NET#745](https://github.com/sshnet/SSH.NET/issues/745)) scp server does not support file time setting

Sorry if I was unclear - the Windows server is changing the file name to include the time and date because Set-SCPItem is trying to overwrite the file.  We don't want the date in there.  In fact, we want the file to be overwritten.  If the destination is set to an empty directory, or a unique file name, the date and time don't show up (desired behavior).

Thanks.

**Comment by MVKozlov** [comment]
to be sure, try using `scp`, like
`scp -p /path/to/filename username@hostname`
The `-p` switch tells the server to save file times
if the command also blocks, then it's the same reason...

**Comment by ChicagoJay** [comment]
> to be sure, try using `scp`, like `scp -p /path/to/filename username@hostname` The `-p` switch tells the server to save file times if the command also blocks, then it's the same reason...

Yep - that locks up too!  So, is there a way to disable that in `Set-SCPItem`?  I see your comment to use SFTP instead.  Is that the only solution?

**Comment by MVKozlov** [comment]
Yes, setting the file time hardcoded in `SSH.NET` library, so SFTP is only solution for now
